### PR TITLE
remove GIT environment vars if any

### PR DESCRIPTION
**Pull request by veksha**
`git_status` addon breaks when CudaText is launched from another app that propagates its GIT env vars.
(CudaText is used as a diff tool in this app)

for example, `Git Extensions` tool pollutes environment with the following vars.

```json
{
    "GIT_DIFFTOOL_NO_PROMPT": "true",
    "GIT_DIFFTOOL_TRUST_EXIT_CODE": "false",
    "GIT_DIFF_PATH_COUNTER": "1",
    "GIT_DIFF_PATH_TOTAL": "1",
    "GIT_DIR": "F:/MySSDPrograms/CudaText_up/src/CudaText/.git",
    "GIT_EXEC_PATH": "C:/Program Files/Git/mingw64/libexec/git-core",
    "GIT_EXTERNAL_DIFF": "git-difftool--helper",
    "GIT_MERGETOOL_GUI": "true",
    "GIT_PAGER": "",
    "GIT_PREFIX": ".",
    "GIT_WORK_TREE": "F:/MySSDPrograms/CudaText_up/src/CudaText"
}
```

in this patch I remove this vars from environment before calling `git` process.
otherwise, git_status will break.

### CODE REVIEW

**Comment by veksha** [comment]
we need to think what more adjustments to make.
as it turns out these env vars come from git process (spawned by Git Extensions) and not from Git Extensions.

we have this situation basically:
1. we launch difftool from `Git Extensions` to compare two files. (CudaText configured as difftool in settings of `Git Extensions`)
2. `Git Extensions` launches `git difftool --tool={path_to_cudatext} ....`
3. CudaText is launched and inherited git environment vars. (see problem1 below)

making CudaText work correctly as a difftool is a little tricky.

problem1: it is dangerous to leave  `"GIT_*" vars` like this, because they will be propagated to every child process launched by Cudatext. we have guarded git_status addon from this (this PR). but other addons are not protected: we can launch git-bash from cuda_exttools, for example, - it will inherit these `"GIT_*" vars`, then any git call from this git-bash will be broken!)
is it ok to strip `"GIT_*" vars` globally (from pascal code)? I think yes. what do you think, @Alexey-T ?

problem 2:
to make CudaText correctly work as a difftool user must also disable `ui_one_instance`:
- because `git difftool` can give relative paths for files. current dir will be set on CudaText process launch. so we need to launch new instance every time and all will be good.

so to sum it up. to make all work correctly we must:
1. strip "git_" vars on CudaText startup
2. disable ui_one_instance

**Comment by Alexey-T** [comment]
1. how my app can strip GIT_ vars?
2. maybe cmd line key helps? `-n - Ignore option "ui_one_instance", and open new app window.`

**Comment by veksha** [comment]
> 2. maybe cmd line key helps? `-n - Ignore option "ui_one_instance", and open new app window`

perfect! didn't know about this. thanks.

> 1. how my app can strip GIT_ vars?

os.environ is a dictionary, you can modify it like this, i think:
`os.environ = { k:v for k,v in os.environ.items() if not k.upper().startswith('GIT_') }`
after initializing python engine.

**Comment by Alexey-T** [comment]
seems 'os' is imported fast
```
user@PC:~$ python3
Python 3.8.10 (default, Nov 14 2022, 12:59:47) 
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import time
>>> t=time.time()
>>> t=time.time() ; import os ; tt=time.time()-t
>>> tt
6.198883056640625e-06
```
so i can add it

**Comment by Alexey-T** [comment]
changing dict is fast 
```
>>> t=time.time() ; os.environ = { k:v for k,v in os.environ.items() if not k.upper().startswith('GIT_') } ; tt=time.time()-t
>>> tt
0.00038552284240722656
```

**Comment by Alexey-T** [comment]
try it first. formmain.pas
```
procedure TfmMain.PythonEngineAfterInit(Sender: TObject);

///add at end
  AppPython.Exec('import os; os.environ = { k:v for k,v in os.environ.items() if not k.upper().startswith("GIT_") }');
end;
```

**Comment by veksha** [comment]
this works:
`AppPython.Exec('import os; exec("for k in os.environ.keys(): k.upper().startswith(\"GIT_\") and os.environ.pop(k)")');`

(`AppPython.Exec` is actually "eval", and doesn't support for loops. so I used "exec" inside of it.)

**Comment by veksha** [comment]
`os.environ.pop` properly triggers env var unset and all now works.

**Comment by Alexey-T** [comment]
but my code above works too, no? I merged it, ok?

**Comment by Alexey-T** [comment]
does this change work?
```pascal
AppPython.Exec('import os; os.environ = { k:v for k,v in os.environ.items() if not k.upper().startswith("GIT_") }; os.environ["GIT"]="d"; os.environ.pop("GIT")');
e
```

**Comment by veksha** [comment]
> does this change work?

no. what is it? :)

**Comment by Alexey-T** [comment]
my last change has new code at and- `os.environ["GIT"]="d"; os.environ.pop("GIT")')` - it don;'t help?

**Comment by veksha** [comment]
I don't understand how it should work. Tried this and it is not working.

**Comment by veksha** [comment]
can you explain the logic behind this, @Alexey-T? i don't get it. `["GIT"]="d"`

**Comment by Alexey-T** [comment]
ok, I got it, i will use your new code.

**Comment by Alexey-T** [comment]
made the fix again. try it.
